Tests/TuistGeneratorTests/Generator/BuildPhaseGeneratorTests.swift

```diff
@@ -632,6 +703,68 @@ final class BuildPhaseGeneratorTests: TuistUnitTestCase {
         ])
     }
 
+    func test_generateResourcesBuildPhase_whenMultiPlatformResourceFiles() throws {
+        // Given
+        let temporaryPath = try temporaryPath()
+        let resources: [ResourceFileElement] = [
+            .file(path: "/Shared.type"),
+            .file(path: "/iOS.type", condition: .when([.ios])),
+            .file(path: "/macOS.type", condition: .when([.macos])),
+            .file(path: "/tvOS.type", condition: .when([.tvos])),
+            .file(path: "/visionOS.type", condition: .when([.visionos])),
+            .file(path: "/watchOS.type", condition: .when([.watchos])),
+        ]
+        let target = Target.test(resources: resources)
+        let fileElements = ProjectFileElements()
+        let pbxproj = PBXProj()
+
+        resources.forEach {
+            let ref = PBXFileReference()
+            pbxproj.add(object: ref)
+            fileElements.elements[$0.path] = ref
+        }
+        let nativeTarget = PBXNativeTarget(name: "Test")
+
+        let graph = Graph.test(path: temporaryPath)
+        let graphTraverser = GraphTraverser(graph: graph)
+        // When
+        try subject.generateResourcesBuildPhase(
+            path: "/path",
+            target: target,
+            graphTraverser: graphTraverser,
+            pbxTarget: nativeTarget,
+            fileElements: fileElements,
+            pbxproj: pbxproj
+        )
+
+        // Then
+        let pbxBuildPhase: PBXBuildPhase? = nativeTarget.buildPhases.first
+        XCTAssertNotNil(pbxBuildPhase)
+        XCTAssertTrue(pbxBuildPhase is PBXResourcesBuildPhase)
+
+        let resourceBuildPhase = try XCTUnwrap(nativeTarget.buildPhases.first as? PBXResourcesBuildPhase)
+        var buildFiles = try XCTUnwrap(resourceBuildPhase.files)
+
+        // sorting to get some determinism in, sort alphabetically, ascending, (uppercase before lowercase)
+        try buildFiles.sort { prev, suc in
```

Comment:
would it be more straight forward to set up `expected` as a dictionary of filenames -> expected platform filters and just do a `for` loop over the `buildFiles` and look up that the filters match?  This would avoid having order matter and if we're doing an individual `XCTAssetEqual` within the `for` loop it would be more granular signal if something failed.

Reply:
Thanks a lot for the suggestion - I added the approach you mentioned now